### PR TITLE
fix(insights): Prevent requesting refreshes of shared dashboards and insights

### PR DESCRIPTION
**Pull request by webjunkie**
## Problem

I found that people can do `refresh=True` on shared insights and dashboards... While they will get fresh results, this always calculates and makes the insight end up being slow.

## Changes

Put in a safeguard 🚧

If stale, it will recalculate asynchronously. This won't show up on the shard insight however for now, you'd have to refresh.
I investigated the frontend logics if we might add polling, but would put this in a follow up since the logics are again complex 🫠

### CODE REVIEW

File: posthog/api/insight.py
Comment (Twixes) [inline posthog/api/insight.py:1]: Would be good to have some test for this, to ensure this stays true in the future

**Review by Twixes — COMMENTED** [review]
Sounds good, just could use a test

**Comment by webjunkie** [comment]
Uh wait something merged wrong here.

**Review by Twixes — APPROVED** [review]
👌

**Pull request merged**